### DISCUSSION TITLE
What data is getting pushed to the cloud?

### DISCUSSION THREAD

**Discussion by tcurdt** [grafana-cold-storage/agent#2671] [Q&A] [answered]
Is there any way to see what the agent is actually pushing out to the Grafana cloud?

While `localhost:12345/metrics` provides information on the agent itself,
there does not seem to be a way to see the final (re-labeled) values that are getting transferred into the cloud.

I was hoping there was an endpoint like `localhost:12345/current`.

**Comment by rfratto** [comment, marked as answer]
There's no way to do this right now. `agentctl` may have some tools for viewing what's in the WAL, but it's probably not exactly what you want. Honestly, the best way to see what's getting pushed to the endpoint is to query the endpoint. 

`localhost:12345/current` is probably not going to be something we'd be able to add, since some agents can have millions of active series, which would be unreasonable to show at that one endpoint.

**Reply by tcurdt** [reply]
Thanks for the pointer to `agentctl`. Is the WAL before or after the processing (re-mappins etc)?

I am not sure that the millions of active series would be such a blocker.
In the end one would use `curl` to download a snapshot of what would leave the agent.
It would make debugging much easier.

But I guess the answer is "not implemented - at least for now".
Thanks for clarifying!
